tests/DCPS/RtpsRelay/Smoke/DataReaderListener.cpp

```diff
@@ -149,8 +149,17 @@ void DataReaderListenerImpl::on_sample_lost(
   ACE_DEBUG((LM_DEBUG, ACE_TEXT("%N:%l: INFO: on_sample_lost()\n")));
 }
 
-bool DataReaderListenerImpl::is_valid() const
+bool DataReaderListenerImpl::is_valid(bool check_lease_recovery) const
```

Comment:
Instead of hoping for certain amounts of data before and after rediscovery, I would examine the BITs to determine when rediscovery happens and just marked that data as being received before and after.  So long as you get some before and some after, the test should pass.